src/test/java/org/terracotta/statistics/derived/latency/DefaultLatencyHistogramStatisticTest.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * All content copyright Terracotta, Inc., unless otherwise indicated.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terracotta.statistics.derived.latency;
+
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.terracotta.statistics.LongSample;
+import org.terracotta.statistics.Time;
+import org.terracotta.statistics.util.ConcurrentParameterized;
+import org.terracotta.statistics.util.LatencyUtils;
+import org.terracotta.statistics.util.Percentile;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import java.util.stream.DoubleStream;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import static java.lang.Math.round;
+import static java.time.Duration.ofMillis;
+import static java.time.Duration.ofMinutes;
+import static java.util.function.DoubleUnaryOperator.identity;
+import static org.hamcrest.Matchers.closeTo;
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.lessThanOrEqualTo;
+import static org.hamcrest.core.IsEqual.equalTo;
+import static org.junit.Assert.assertThat;
+import static org.terracotta.statistics.util.LatencyUtils.getPercentiles;
+
+/**
+ * @author Mathieu Carbou
+ */
+@RunWith(ConcurrentParameterized.class)
+public class DefaultLatencyHistogramStatisticTest {
```

Comment:
This feels like it's really a test of the `BarSplittingBiasedHistogram` class?  Shouldn't any missing test coverage be moved to a unit test for that class?

Reply:
I did this to test `DefaultLatencyHistogramStatistic`, which works on Long and does some logic with expire) with my dataset)... I've found some issues in `BarSplittingBiasedHistogram` with that but I think you added the coverage after ?
Let me know which tests are missing from `BarSplittingBiasedHistogram` I'll try to add them.